### PR TITLE
Enable authentication for firebase RTDB

### PR DESCRIPTION
**Pull request by austintjy**
Generates an access token using Google's REST authentication endpoint (email authentication)
Attaches the access token to every request thereafter using the "auth" url parameter

See https://firebase.google.com/docs/reference/rest/database/
See https://firebase.google.com/docs/reference/rest/auth

### CODE REVIEW

**Comment by ckoever** [comment]
ctx = ussl.SSLContext()

SSLContext is not supported by ussl (at least on micropython)

Where can i download a pycopy bin for my esp32 ?

File: ufirebase.py [hunk 1]
```diff
@@ -3,10 +3,16 @@
 import ussl
 import _thread
 import time
+import urllib.urequest
```
Comment (ckoever) [inline ufirebase.py]: SSLContext not sup. in urequest
see ssl.SSLContext()](https://docs.micropython.org/en/latest/library/ssl.html)

**Review by ckoever — CHANGES_REQUESTED** [review]
SSLContext() alternative

File: ufirebase.py (hunk 1 above)
Comment (austintjy) [reply ufirebase.py]: urllib.urequest is from pycopy https://github.com/pfalcon/pycopy-lib

I am using a slightly modified version of the urequest due to the SSLContext()

Will commit it too, i completely overlooked that

Comment (austintjy) [reply ufirebase.py]: Just a thing to note is that I am not using the full pycopy lib as space on my ESP32 is limited, i am only specifically using urllib.urequest and modified it for my own purposes.

For my purposes the workaround for the SSLContext issue works fine, might want to opt for a permanent solution or a fork of pycopy with the same modifications if you are interested in using the entire library

**Comment by ckoever** [comment]
I will finish the readme in the next few days.
If I find the time (school) I will also create a real wiki for the project.

**Review by ckoever — APPROVED** [review]

**Comment by austintjy** [comment]
> 

Alright cool, thanks for the merge/overhaul.

**Pull request merged**